sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -271,13 +272,21 @@ private HttpRequest createHttpRequest(SwaggerMethodParser methodParser, Object[]
     @SuppressWarnings("unchecked")
     private HttpRequest configRequest(final HttpRequest request, final SwaggerMethodParser methodParser,
         final Object[] args) throws IOException {
-        final Object bodyContentObject = methodParser.setBody(args);
+
+        final Object bodyContentObject;
+        String contentType = methodParser.getBodyContentType();
+
+        if (ContentType.MULTIPART_FORM_DATA.equals(contentType)) {
+            String boundary = UUID.randomUUID().toString();
+            bodyContentObject = methodParser.setMultipartBody(args, boundary);
+            contentType += "; boundary=" + boundary;
+        } else {
+            bodyContentObject = methodParser.setBody(args);
+        }
+
         if (bodyContentObject == null) {
             request.getHeaders().set("Content-Length", "0");
         } else {
-            // We read the content type from the @BodyParam annotation
-            String contentType = methodParser.getBodyContentType();
-
```

Comment:
In a small way I dislike seeing code move onto the critical path more than it was before. Are you able to more eagerly short-circuit if, e.g. `args` is empty / null?

Reply:
I'll rewrite how we do things in `SwaggerMethodParser` so we only have one method (`setBody()`), this way we don't need to move logic around here.